magma/syntax/util.py

```diff
@@ -23,14 +42,23 @@ def build_call_args(io, annotations):
     return [getattr(io, param) for param in annotations if param != "return"]
 
 
-def wire_call_result(io, call_result, annotations):
+def wire_call_result(io, call_result, annotations, output_port_names=None):
     if isinstance(call_result, Circuit):
         if not len(call_result.interface.outputs()) == 1:
             raise TypeError(
                 "Expected register return instance with one output")
         call_result = call_result.interface.outputs()[0]
     if isinstance(annotations["return"], tuple):
-        for i in range(len(annotations["return"])):
-            wire(getattr(io, f"O{i}"), call_result[i])
+        if output_port_names is None:
+            for i in range(len(annotations["return"])):
+                wire(getattr(io, f"O{i}"), call_result[i])
+        else:
+            for i, name in enumerate(output_port_names):
+                wire(getattr(io, name), call_result[i])
+
+
     else:
-        wire(io.O, call_result)
+        if output_port_names is None:
+            wire(io.O, call_result)
+        else:
+            wire(getattr(io, output_port_names[0]), call_result)
```

Comment:
Hmm, I guess if we fill output_port_names then we avoid the if statements here too (just use output_port_names, which will have the default values for None)

Reply:
same issue as above.